starport_template/lib/pages/assets_portfolio_page.dart

```diff
@@ -101,7 +112,9 @@ class _AssetsPortfolioPageState extends State<AssetsPortfolioPage> {
       Navigator.of(context).push(MaterialPageRoute(builder: (context) => const TransactionHistoryPage()));
 
   Future _fetchWalletBalances() async {
-    await StarportApp.walletsStore.getBalances(selectedWallet.publicAddress);
+    if (widget.shouldFetchBalances == true) {
```

Comment:
```suggestion
    if (widget.shouldFetchBalances) {
```

also, this won't work, since the number of different cases where you open the `AssetsPortfolioPage` can be multiple and it can grow later on. it's not the page responsibility to determine whether new balances should be fetched or not. We can optimize the process by simply removing the `StarportApp.walletsStore.getBalances` from here since balances are loaded automatically when the current wallet is being changed ([here](https://github.com/tendermint/flutter/blob/19c6b3bd5bd0e7a29834bc5303a0a5903aa98e7a/starport_template/lib/stores/wallets_store.dart#L106)). Which happens on the app start and each manual change of the wallet. 

We can also trigger `getBalances` upon successful transaction broadcast aswell (I'll create separate issue for that).

All in all, getting balances is not the root cause here and even with this implementation, the bug persists. We've discovered it happens most of the time when importing the first wallet that already has some tokens on the fresh app installation.

Reply:
I see, my bad, will fix this up